internal/controller/policyserver_controller_test.go

```diff
@@ -477,74 +287,62 @@ var _ = Describe("PolicyServer controller", func() {
+			createPolicyServerAndWaitForItsService(ctx, policyServer)
+
 			Consistently(func() error {
-				_, err := getPolicyServerPodDisruptionBudget(policyServerName)
+				_, err := getPolicyServerPodDisruptionBudget(ctx, policyServerName)
 				return err
 			}, consistencyTimeout, pollInterval).ShouldNot(Succeed())
 		})
 
-		It("should create the PolicyServer pod with the limits and the requests", func() {
+		It("should create the PolicyServer deployment with the limits and the requests", func() {
 			policyServer := policyServerFactory(policyServerName)
 			policyServer.Spec.Limits = corev1.ResourceList{
 				"cpu":    resource.MustParse("100m"),
 				"memory": resource.MustParse("1Gi"),
 			}
-			createPolicyServerAndWaitForItsService(policyServer)
+			createPolicyServerAndWaitForItsService(ctx, policyServer)
+
 			By("creating a deployment with limits and requests set")
 			Eventually(func() error {
-				deployment, err := getTestPolicyServerDeployment(policyServerName)
+				deployment, err := getTestPolicyServerDeployment(ctx, policyServerName)
 				if err != nil {
 					return err
 				}
 				Expect(deployment.Spec.Template.Spec.Containers[0].Resources.Limits).To(Equal(policyServer.Spec.Limits))
 				return nil
 			}, timeout, pollInterval).Should(Succeed())
-
-			By("creating a pod with limit and request set")
-			Eventually(func() error {
-				pod, err := getTestPolicyServerPod(policyServerName)
-				if err != nil {
-					return err
-				}
-
-				Expect(pod.Spec.Containers[0].Resources.Limits).To(Equal(policyServer.Spec.Limits))
-
-				By("setting the requests to the same value as the limits")
-				Expect(pod.Spec.Containers[0].Resources.Requests).To(Equal(policyServer.Spec.Limits))
-
-				return nil
-			}, timeout, pollInterval).Should(Succeed())
 		})
 
 		It("should create deployment with owner reference", func() {
 			policyServer := policyServerFactory(policyServerName)
-			createPolicyServerAndWaitForItsService(policyServer)
+			createPolicyServerAndWaitForItsService(ctx, policyServer)
+
 			Eventually(func() error {
-				deployment, err := getTestPolicyServerDeployment(policyServerName)
+				deployment, err := getTestPolicyServerDeployment(ctx, policyServerName)
 				if err != nil {
 					return err
 				}
-				policyServer, err := getTestPolicyServer(policyServerName)
+				policyServer, err := getTestPolicyServer(ctx, policyServerName)
 				if err != nil {
 					return err
 				}
 				Expect(deployment.OwnerReferences).To(ContainElement(
 					MatchFields(IgnoreExtras, Fields{
-						"UID":        Equal(policyServer.GetUID()),
-						"Name":       Equal(policyServer.GetName()),
-						"Kind":       Equal(policyServer.GetObjectKind().GroupVersionKind().Kind),
-						"APIVersion": Equal(policyServer.GetObjectKind().GroupVersionKind().GroupVersion().String()),
+						"UID":  Equal(policyServer.GetUID()),
+						"Name": Equal(policyServer.GetName()),
+						// "Kind":       Equal(policyServer.GetObjectKind().GroupVersionKind().Kind),
```

Comment:
Add TODO explaining why we had to comment these lines, maybe also create an issue so that we don't forget about it

Reply:
there is a FIXME in the other assertions but I forgot to add it here

Reply:
This was already tracked by #792